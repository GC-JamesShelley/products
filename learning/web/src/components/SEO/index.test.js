import React from "react"
import SEO from "./index"
import { shallow } from "enzyme"
import * as Gatsby from "gatsby"

describe(SEO, () => {
  it("should render", () => {
    const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          author: "Bob Kane",
          description:
            "All brand new adventures of THE BATMAN and ROBIN, the boy wonder",
          title: "Detective Comics #27",
        },
      },
    }))
    const component = shallow(
      <SEO description={"Amazing"} lang={"en"} title={"ðŸ˜€"} />
    )
    expect(component).toMatchSnapshot()
  })
})
