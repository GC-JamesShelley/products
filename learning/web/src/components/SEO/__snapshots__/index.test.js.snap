// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO should render 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en",
    }
  }
  meta={
    Array [
      Object {
        "content": "Amazing",
        "name": "description",
      },
      Object {
        "content": "Bob Kane",
        "name": "author",
      },
      Object {
        "content": "ðŸ˜€",
        "property": "og:title",
      },
      Object {
        "content": "Amazing",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "noindex, no follow",
        "name": "robots",
      },
    ]
  }
  title="ðŸ˜€"
  titleTemplate="%s | Detective Comics #27"
/>
`;
